server/src/main/java/com/epam/aidial/core/server/Proxy.java

```diff
@@ -280,6 +281,11 @@ private void respond(HttpServerRequest request, HttpStatus status) {
     }
 
     private void respond(HttpServerRequest request, HttpStatus status, String body) {
+        if (status == HttpStatus.OK) {
+            SpanUtil.updateName(request.method().name(), request.path());
+        } else if (status.is4xx() || status.is5xx()) {
```

Comment:
Why not use if/else?

what if http 3xx comes?

Reply:
Will be fixed in scope of a separate issue